fabtests/unit/mr_cache_evict.c

```diff
@@ -0,0 +1,695 @@
+/*
+ * (C) Copyright 2020 Hewlett Packard Enterprise Development LP
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#include <unistd.h>
+#include <stddef.h>
+#include <sys/mman.h>
+#include <getopt.h>
+#include <time.h>
+
+#ifdef __linux__
+#include <malloc.h>
+#endif
+
+#include "unit_common.h"
+#include "shared.h"
+
+/* Supported memory region types. */
+enum mem_type {
+	MMAP,
+	BRK,
+	SBRK,
+};
+
+struct memory_region {
+	enum mem_type type;
+	size_t size;
+	char *user;
+};
+
+/* Given a time value in seconds, determine the expected cached time value. The
+ * assumption is that the cache value should at least have a
+ * CACHE_IMPROVEMENT_PERCENT improvement over the original value.
+ */
+#define CACHE_IMPROVEMENT_PERCENT 0.8
+#define CACHE_TIME_MAX_VALUE(time) ((time) * (1 - CACHE_IMPROVEMENT_PERCENT))
+
+#define PGMAP_PRSNT (1UL << 63)
+#define PFN_MASK ((1UL << 55) - 1)
+#define FOFFSET(va) ((((uint64_t)va) >> 12) << 3)
+
+static void *reuse_addr = NULL;
+static char err_buf[512];
+static size_t mr_buf_size = 4097;
```

Comment:
This should be controlled via input.  I would use a larger default size than something less than a page.  Is there a reason that was selected?

Reply:
I'll set the default large (doesn't mater to me what it is). In addition, I added a "-s" argument to change the buffer size.